### PR TITLE
feat(stdlib/experimental): count distinct by tag

### PR DESCRIPTION
**Pull request by kelwang**
close https://github.com/influxdata/idpe/issues/6192

### Done checklist
- [ ] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: stdlib/experimental/count/count.flux [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import "experimental"
+
+distinctByTag = (tables=<-, measurement, tag) =>
+	tables
+		|> filter(fn: (r) => r._measurement == measurement)
+		|> keep(columns: [tag])
+  		|> limit(n: 1)
+  		|> group()
+  		|> count(column: tag)
+		|> yield()
```
Comment (jsteenb2) [inline stdlib/experimental/count/count.flux]: out of curiosity, why is the yield needed here?

Comment (kelwang) [reply stdlib/experimental/count/count.flux]: the error message told me so, I am not sure why either

Comment (jsteenb2) [reply stdlib/experimental/count/count.flux]: hmm, hopefully @nathanielc will know 🤞

**Comment by sanderson** [comment]
Should this be included as part of the `experimental/aggregate` package rather than the base `experimental` package?

**Comment by kelwang** [comment]
> Should this be included as part of the `experimental/aggregate` package rather than the base `experimental` package?

good idea

**Comment by nathanielc** [comment]
Closing as stale, but would accept a change like this in the future.

**Pull request closed without merging**